sql-plugin/src/main/java/com/nvidia/spark/RapidsUDF.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark;
+
+import ai.rapids.cudf.ColumnVector;
+
+/** A RAPIDS-accelerated version of a user-defined function (UDF). */
+public interface RapidsUDF {
+  /**
+   * Evaluate a user-defined function with RAPIDS cuDF columnar inputs
+   * producing a cuDF column as output.
+   * <p>
+   * Note that the inputs should NOT be closed by this method, as they will
+   * be closed by the caller. This method must close any intermediate cuDF
+   * results produced during the computation (e.g.: `Table`, `ColumnVector`
+   * or `Scalar` instances).
+   * @param args columnar inputs to the UDF that will be closed by the caller
+   *             and should not be closed within this method.
+   * @return columnar output from the user-defined function
```

Comment:
Do we want/need to add anything about what happens if the type of the output does not match the expected UDF type?  We end up throwing an exception.  The main reason I call this out is because `UDF` is done through reflection so the type is backed in, but cudf is type erased so we don't get the same static feedback.